src/main/kotlin/com/ampnet/reportserviceeth/blockchain/TransactionInfo.kt

```diff
@@ -95,4 +96,36 @@ data class TransactionInfo(
         asset?.name,
         asset?.symbol
     )
+
+    /*
+     * from - address of the PayoutManager contract.
+     * to - address of the wallet which received revenue share payout.
+     */
+    constructor(
+        event: TransactionEvents.ReleaseEventResponse,
+        txRecipient: TransactionReceipt,
+        asset: IAsset.AssetState?
```

Comment:
Looks to me that `IAsset.AssetState` doesn't need to be nullable because you use the function that returns a non-nullable value.